smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/KafkaSinkWithLegacyMetadataTest.java

```diff
@@ -0,0 +1,112 @@
+package io.smallrye.reactive.messaging.kafka;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+import org.apache.kafka.common.serialization.IntegerDeserializer;
+import org.apache.kafka.common.serialization.IntegerSerializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.reactivestreams.Publisher;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.reactive.messaging.kafka.base.KafkaMapBasedConfig;
+import io.smallrye.reactive.messaging.kafka.base.KafkaTestBase;
+import io.smallrye.reactive.messaging.kafka.impl.KafkaSink;
+
+/**
+ * Duplicate of {@link KafkaSinkTest} - delete once we remove the legacy
+ * {@link io.smallrye.reactive.messaging.kafka.api.OutgoingKafkaRecordMetadata}
+ * <p/>
+ * Tests not using metadata have been removed
```

Comment:
The honest reason for this is that when keeping all the test methods, I got problems in one of the `testABeanProducingXXXX()` methods (I forgot which). The problem is that the `expected` AtomicInteger contained 20 rather than 10 entries.

It worked fine on its own, so it is probably some test setup pollution. I didn't try to track down the issue since I realised it didn't have anything to do with metadata

Reply:
Make sure to use a unique topic.